src/main/kotlin/org/magrathea/slack/service/SlackService.kt

```diff
@@ -1,16 +1,28 @@
 package org.magrathea.slack.service
 
 import com.github.seratch.jslack.Slack
+import com.github.seratch.jslack.api.methods.MethodsClient
+import org.magrathea.slack.config.SlackConfigurationProperties
 import org.magrathea.slack.interactor.SlackSetDoNotDisturbInteractor
+import org.magrathea.slack.interactor.SlackStopDoNotDisturbInteractor
 import javax.enterprise.context.ApplicationScoped
 import javax.enterprise.inject.Default
 import javax.inject.Inject
 
-@Default
 @ApplicationScoped
-open class SlackService(@Inject private val slack: Slack) : ISlackService {
+open class SlackService(slack: Slack, slackConfigurationProperties: SlackConfigurationProperties) : ISlackService {
 
-    override fun setDoNotDisturb(token: String) {
-        SlackSetDoNotDisturbInteractor.execute(slack.methods(token))
+    var methodsClient: MethodsClient? = null
+
+    init {
+        this.methodsClient = slack.methods(slackConfigurationProperties.oAuthToken)
+    }
```

Comment:
This can be simplified:

```kotlin
private val methodsClient = slack.methods(slackConfigurationProperties.oAuthToken)
```

Which also removes the need for the null checks.